src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -85,11 +83,9 @@ public static void SetCurrentCircuitHost(CircuitHost circuitHost)
 
         public IServiceProvider Services { get; }
 
-        public CircuitSynchronizationContext SynchronizationContext { get; }
-
         public async Task InitializeAsync()
         {
-            await SynchronizationContext.Invoke(() =>
+            await Renderer.Invoke(() =>
```

Comment:
Just use Renderer.Invoke as it will internally call SynchronizationContext.Invoke and that way there's no need for the SyncContext to be explicit

Reply:
I passed in the synccontext as a parameter, but kept using it through the renderer. Introduced an IDispatcher interface for this.